config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/string/CharacterBlacklistUtil.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.util.string;
+
+import java.util.Set;
+
+import static com.ca.apim.gateway.cagatewayconfig.util.environment.CharacterBlacklist.getCharBlacklist;
+
+/**
+ * Utility class for Character Blacklist. Purpose is to apply functions to the constants found in CharacterBlacklist
+ */
+public class CharacterBlacklistUtil {
+
+    /**
+     * Find blacklist characters and replace with value '-'
+     *
+     * @param string with invalid characters
+     * @return a string stripped of all invalid characters and replaced with value '-'
+     */
+    public static String filterAndReplace(String string) {
+        Set<Character> charBlacklist = getCharBlacklist();
+        final char replacementChar = '-';
+        StringBuilder processedString = new StringBuilder();
+
+        for (char c : string.toCharArray()) {
```

Comment:
can't you use String.replaceall?
if not, use Stream here, Stream.of(string.tochararray).foreach

Reply:
String.replace will work here. Replaceall requires a double escape for backslashes. I think for visual simplicity using replace will be fine.